plutus-tx-plugin/test/Plugin/Spec.hs

```diff
@@ -312,6 +313,15 @@ ptreeConstruct = plc @"ptreeConstruct" (Two (Two (One ((1,2),(3,4)))) :: B Int)
 ptreeMatch :: CompiledCode (B Int -> Int)
 ptreeMatch = plc @"ptreeMatch" (\(t::B Int) -> case t of { One a -> a; Two _ -> 2; })
 
+polyRec :: CompiledCode (B Int -> Int)
+polyRec = plc @"polyRec" (
+    let
+        depth :: B a -> Int
+        depth tree = case tree of
+            One _     -> 1
+            Two inner -> 1 + depth inner
+    in \(t::B Int) -> depth t)
+
```

Comment:
Perhaps also add the test that I wrote a while ago, but did not include? This one:

```haskell
ptreeMatch :: CompiledCode (B Int -> Int)
ptreeMatch = plc @"ptreeMatch" (
    let go :: (a -> Int) -> B a -> Int
        go _ k (One x) = k x
        go _ k (Two b) = go () (\(x, _) -> k x) b
    in go () (\x -> x))
```

Reply:
Yep.